src/Components/Pokedex/Pokedex.test.js

```diff
@@ -1,6 +1,117 @@
-import React from 'react'
-import Pokedex from './Pokedex'
-import { render, fireEvent, waitFor } from "@testing-library/react";
+import React from 'react';
+import Pokedex from './Pokedex';
+import { render, act } from "@testing-library/react";
+import {MemoryRouter} from 'react-router-dom';
 import "@testing-library/jest-dom";
+import userEvent from '@testing-library/user-event';
+import {getSinglePokemon} from '../../ApiCalls';
 
-const pokemonList = {}
+jest.mock('../../ApiCalls')
+
+const pokemonList = [
+  {
+    "name": "ivysaur",
+    "url": "https://pokeapi.co/api/v2/pokemon/2/"
+  },
+  {
+    "name": "venusaur",
+    "url": "https://pokeapi.co/api/v2/pokemon/3/"
+  },
+  {
+    "name": "charmander",
+    "url": "https://pokeapi.co/api/v2/pokemon/4/"
+  },
+  {
+    "name": "charmeleon",
+    "url": "https://pokeapi.co/api/v2/pokemon/5/"
+  },
+  {
+    "name": "squirtle",
+    "url": "https://pokeapi.co/api/v2/pokemon/7/"
+  },
+  {
+    "name": "wartortle",
+    "url": "https://pokeapi.co/api/v2/pokemon/8/"
+  }
+]
+
+describe("Pokedex component", () => {
+  it('should have a select menu', () => {
+    const { getByRole, getAllByRole } = render(
+      <MemoryRouter>
+        <Pokedex allPokemon={pokemonList}/>
+      </MemoryRouter>)
+
+    const pokemonMenu = getByRole('combobox', {name:'Pokemon List'})
+    const pokemonListOptions = getAllByRole('option')
+
+    expect(pokemonMenu).toBeInTheDocument();
+    expect(pokemonListOptions.length).toEqual(6)
+  })
+})
+
+describe("User interactions through Pokedex", () => {
+
+  getSinglePokemon.mockResolvedValueOnce({
+      name: "venusaur",
+      height: 20,
+      weight: 1000,
+      sprites: {
+        front_default: "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/3.png"
+      }
+    })
+
+  it("should pick and display different pokemon's info", async () => {
+    const { getByRole, findByRole } = render(
+      <MemoryRouter>
+        <Pokedex allPokemon={pokemonList}/>
+      </MemoryRouter>)
+
+    const pokemonMenu = getByRole('combobox', {name:'Pokemon List'})
+
+    act(() => {
+      userEvent.selectOptions(pokemonMenu, "https://pokeapi.co/api/v2/pokemon/3/")
+    })
+
+    expect(await pokemonMenu.value).toEqual("https://pokeapi.co/api/v2/pokemon/3/")
+    expect(getSinglePokemon).toHaveBeenCalledTimes(1)
+    expect(getSinglePokemon).toHaveBeenCalledWith("https://pokeapi.co/api/v2/pokemon/3/")
+
+    let pokemonImg = await findByRole('img', {name: 'venusaur'});
```

Comment:
So if you use findByRole, you no longer need jest --sixteen?

Reply:
I was pretty curious about that. Even if jest --sixteen the await waitFor(() => {}) wasn't quite doing it for my test. Super strange since I thought they were the same thing.